crates/aptos-crypto/src/bls12381/bls12381_sigs.rs

```diff
@@ -106,11 +106,15 @@ impl Signature {
         msgs: &[&T],
         pks: &[&PublicKey],
     ) -> Result<()> {
-        let msgs = msgs
+        let mut messages: Vec<Vec<u8>> = vec![];
+        for message in msgs {
+            messages.push(signing_message(*message)?);
+        }
+
+        let msgs_refs = messages
             .iter()
-            .map(|&m| signing_message(m))
```

Comment:
Putting the `?` operator here wouldn't have worked?

Reply:
You can't call ? from within a `map` as it's in a different scope.

Reply:
Same as above